### PR TITLE
Added new feature abundance vs velocity plot

### PR DESCRIPTION
**Pull request by sandsinghal**
This is a PR to solve the first objective of the project titled "Custom Abundance Widget" for GSOC 2021
## Description

The objective required creation of a plot of abundance of elements as a function of velocity. I have implemented it using two functions. First function (`plot_abundance_vs_velocity`) plots with the help of `matplotlib` and second function (`plotly_abundance_vs_velocity`) plots using `plotly`. I have made two changes in the main code to do the same:
* Created a file `abundance_vs_velocity.py` in the location `tardis\visualization\widgets\abundance_vs_velocity.py`
* Imported the definition `plot_abundance_vs_velocity` in `tardis\visualization\__init__.py`
* Imported the definition `plotly_abundance_vs_velocity` in `tardis\visualization\__init__.py`

What does it require to obtain a plot?
* Need to create a `.yml` format file or a 'csvy model` format file
* Run the simulation on the created file
* Pass the obtained simulation to the created function `plot_abundance_vs_velocity`

To run the plot using matplotlib, need to run the following commands:
`from tardis import run_tardis`
`sim = run_tardis('any_example_file.yml')`
`from tardis.visualization import plot_abundance_vs_velocity`
`plot_abundance_vs_velocity(sim)`

To run the plot using plotly, need to run the following commands:
`from tardis import run_tardis`
`sim = run_tardis('any_example_file.yml')`
`from tardis.visualization import plotly_abundance_vs_velocity`
`plotly_abundance_vs_velocity(sim)`

Could the respective mentors provide some feedback on this task @jaladh-singhal
## How Has This Been Tested?
- [ ] Testing pipeline
- [ ] Reference Data Comparison following these [instructions](https://tardis-sn.github.io/tardis/development/continuous_integration.html)
- [ ] Other (please describe)

## Screenshots (if appropriate):
Ran on the file radiative_csvy.yml with some modeification made. I have added few more velocity points to increase the shells from 2 to 6. 
I obtained the following result using matplotlib implementation.
Screenshot from 2021-03-17 21-11-35

I obtained the following result using plotly implementation.
Screenshot from 2021-03-18 23-38-56

Link to the csvy model format file used is [here](https://drive.google.com/drive/folders/12Qjf9ijbcAFPLh6eMij7IUaPQyOOi0fo?usp=sharing).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] None of the above (please describe)

## Checklist:

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have built the documentation on my fork following [these instructions](https://tardis-sn.github.io/tardis/development/documentation_preview.html)
- [ ] I have assigned and requested two reviewers for this pull request

### CODE REVIEW

**Comment by sandsinghal** [comment]
I am getting codecov/project error Merging #1488 (03c6a11) into master (9497250) will decrease coverage by 1.38%. Could somebody tell me how to resolve this problem.

**Comment by jamesgillanders** [comment]
Can you add velocity units to the x-axis?

**Comment by sandsinghal** [comment]
> Can you add velocity units to the x-axis?

I have added the units and updated the code. Please check now.

**Comment by marxwillia** [comment]
@sandsinghal great work!  Can you please make another plotting function that makes the same plot, but instead uses the interactive python plotting tool plotly?

**Comment by sandsinghal** [comment]
I have added a new plotting function `plotly_abundance_vs_velocity` in the same file `abundance_vs_velocity.py`. This function plots with the help of plotly. Please check now.

File: tardis/visualization/widgets/abundance_vs_velocity.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+            "x": 0.5,
+            "xanchor": "center",
+            "yanchor": "top",
+        },
+    )
+    fig.update_yaxes(range=[0, 1])
+    fig.show()
+
+
+def atomic_number_to_element_symbol():
```
Comment (jaladh-singhal) [inline tardis/visualization/widgets/abundance_vs_velocity.py:118]: We already have similar function in tardis.util - you can even find other visualization modules using it

File: tardis/visualization/widgets/abundance_vs_velocity.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+            go.Scatter(
+                x=velocity[:-1],
+                y=df[index],
+                mode="lines+markers",
+                name=a[index],
+            )
+        )
+    fig.update_layout(
+        xaxis=dict(
+            title="Velocity in " + str(sim.model.velocity.unit),
```
Comment (jaladh-singhal) [inline tardis/visualization/widgets/abundance_vs_velocity.py:102]: Format strings will be better :)

File: tardis/visualization/widgets/abundance_vs_velocity.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+    # obtaing a map of atomic numbers to element symbol
+    # the index of a will be the atomic number and corresponding value will be element symbol
+    a = atomic_number_to_element_symbol()
+
+    # plotting the graph of abundance of element vs velocity
+
+    # for loop iterate and plot abundance of each element at a time
+    fig = go.Figure()
+    for index, row in df.T.iterrows():
```
Comment (jaladh-singhal) [inline tardis/visualization/widgets/abundance_vs_velocity.py:91]: Are you sure this is the most time-efficient way to iterate over a pandas dataframe?

**Review by jaladh-singhal — CHANGES_REQUESTED** [review]
@sandsinghal Sorry for such delayed reply. This looks great except few comments I've mentioned.

**Pull request closed without merging**